ls1intum/Artemis
# Feature/tutor leaderboard view

### Checklist

### Motivation and Context
The tutor leaderboard took >30s to get calculated and was not visible to tutors.
Also the tutor leaderboard only counted the number of assessments (without weighing the complexity of the exercise) and without taking complaints into account.

### Description
I added readonly database views to calculate the tutor leaderboard in < 1 s. The tutor leaderboard now includes the exercise max points for each assessment so that an assessment for an exercise with 8 points is weighed 4 times more than an assessment for an exercise with 2 points. Accepted complaints lead to negative points and complaint responses also give points.

The tutor dashboard for the whole course and for each exercise is calculated like this and is now blazingly fast ;-) and available to instructors and tutors.

### Steps for Testing
1. Log in to ArTEMiS
2. Navigate to Course Administration
3. Navigate to Instructor Course Dashboard --> check the tutor leaderboard for the whole course
4. Navigate to one Instructor Exercise Dashboard --> check the tutor leaderboard for one exercise
5. Navigate to Tutor Course Dashboard --> check the tutor leaderboard for the whole course
6. Navigate to one Tutor Exercise Dashboard --> check the tutor leaderboard for one exercise

### Screenshots
![image](https://user-images.githubusercontent.com/744067/59977682-9d873500-95d4-11e9-978b-e83615301e19.png)
![image](https://user-images.githubusercontent.com/744067/59977685-b263c880-95d4-11e9-81d0-cc81af812fc7.png)
